[3.0] Allows members to see own login history and profile edit history

Sesquipedalian opened the pull request:
Fixes #7790 

With this change, members can now see the history of logins and the history of profile edits for their own accounts. The permissions that allow a member to edit their login info are the same ones that allow the member to view their login history. Likewise, the permission that allows a member to edit their forum profile is the same one that allows the member to view the history of edits to their profile.

This PR also fixes a number of smaller issues I found in related code.

Sesquipedalian committed:
Fixes some typos

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

Sesquipedalian committed:
Fixes "Profile of {name}" in linktree and summary page title

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

Sesquipedalian committed:
Adds support for own/any permissions in Profile subsections

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

Sesquipedalian committed:
Simplifies sub-actions & permission checks in Actions\Profile\Tracking

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

Sesquipedalian committed:
Allows members to see own login history and profile edit history

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

Sesquipedalian committed:
Fixes display of IP addresses in user tracking

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

Sesquipedalian committed:
Shows translated language strings, not tokens, in profile edit history

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

Sesquipedalian committed:
Fixes error about undefined 'tabs' key on profile page user can't view

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

Sesquipedalian committed:
Fixes issues logging group membership changes

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

jdarwood007 commented:
```
SMF\Lang::tokenTxtReplace(): Argument #1 ($string) must be of type string, int given, called in /Sources/Actions/Profile/Tracking.php on line 775
```

Not really related to the PR.  I would fix it by allowing `tokenTxtReplace` to accept an int and then cast it to string.

```
: Trying to access array offset on null in
/Sources/Actions/Profile/Tracking.php
on line
723
```
Fixed it by null coalescing that and the next line.  Interesting data, not edited:
<img width="1210" height="167" alt="Screen Shot 2025-08-24 at 8 44 03 AM" src="https://github.com/user-attachments/assets/492815ae-3f33-4626-9bee-b556c9235e5d" />

Additionally, if an admin changes their additional member groups and views the profile edits, they don't see this.  Admins should still see hidden/invisible group changes.

Sesquipedalian committed:
Allows Lang::tokenTxtReplace() to accept ints

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

Sesquipedalian committed:
Lets admins & group mods see invisible groups in profile edit history

Signed-off-by: Jon Stovell <jonstovell@gmail.com>

Sesquipedalian commented:
> ```
> SMF\Lang::tokenTxtReplace(): Argument #1 ($string) must be of type string, int given, called in /Sources/Actions/Profile/Tracking.php on line 775
> ```
> 
> Not really related to the PR. I would fix it by allowing `tokenTxtReplace` to accept an int and then cast it to string.

Fixed in 9c5e8d5


> ```
> : Trying to access array offset on null in
> /Sources/Actions/Profile/Tracking.php
> on line
> 723
> ```
> 
> Fixed it by null coalescing that and the next line. 

Fixed in 43f82cd.

> Interesting data, not edited: <img alt="Screen Shot 2025-08-24 at 8 44 03 AM" width="1210" height="167" src="https://private-user-images.githubusercontent.com/732116/481391061-492815ae-3f33-4626-9bee-b556c9235e5d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYwNjEwNTMsIm5iZiI6MTc1NjA2MDc1MywicGF0aCI6Ii83MzIxMTYvNDgxMzkxMDYxLTQ5MjgxNWFlLTNmMzMtNDYyNi05YmVlLWI1NTZjOTIzNWU1ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyNFQxODM5MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iODRkNmNhYjRkMWU4YTBlMTllMjA2N2U0ODFiYzRkODdkOTdkZDYzYjJmOWY1ZTEwZGEzY2JiNjJiZTg0NDJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.cKS03O2umjP3-gv-i6HKc4hEx7m0MC3w37VWtaLnz8I">

Weird, but out of scope for this PR.

> Additionally, if an admin changes their additional member groups and views the profile edits, they don't see this. Admins should still see hidden/invisible group changes.

Fixed in 43f82cd.

Sesquipedalian committed:
Merge branch 'release-3.0' into 3.0/track_own_history
Signed-off-by: Jon Stovell <jonstovell@gmail.com>
